framework/src/modules/chain/submodules/loader.js

```diff
@@ -255,96 +239,62 @@ __private.loadSignatures = function(cb) {
  * @todo Add description for the params
  * @todo Missing error propagation when calling balancesSequence.add
  */
-__private.loadTransactions = function(cb) {
-	async.waterfall(
-		[
-			function(waterCb) {
-				self.getRandomPeerFromNetwork((getRandomPeerFromNetworkErr, peer) => {
-					if (getRandomPeerFromNetworkErr) {
-						return setImmediate(waterCb, getRandomPeerFromNetworkErr);
-					}
-					return setImmediate(waterCb, null, peer);
-				});
-			},
-			function(peer, waterCb) {
-				library.logger.info(`Loading transactions from: ${peer.string}`);
-				peer.rpc.getTransactions((err, res) => {
-					if (err) {
-						modules.peers.remove(peer);
-						return setImmediate(waterCb, err);
-					}
-					return library.schema.validate(
-						res,
-						definitions.WSTransactionsResponse,
-						validateErr => {
-							if (validateErr) {
-								return setImmediate(waterCb, validateErr[0].message);
-							}
-							return setImmediate(waterCb, null, peer, res.transactions);
-						}
-					);
-				});
-			},
-			function(peer, transactions, waterCb) {
-				async.eachSeries(
-					transactions,
-					(transaction, eachSeriesCb) => {
-						const id = transaction ? transactions.id : 'null';
-
-						try {
-							transaction = library.logic.transaction.objectNormalize(
-								transaction
-							);
-						} catch (e) {
-							library.logger.debug('Transaction normalization failed', {
-								id,
-								err: e.toString(),
-								module: 'loader',
-								transaction,
-							});
-
-							library.logger.warn(
-								`Transaction ${id} is not valid, peer removed`,
-								peer.string
-							);
-							modules.peers.remove(peer);
+__private.getTransactionsFromNetwork = async function() {
```

Comment:
i don't think this can be `async` because it's used in `async` library which expects callback style